include/mbedtls/error.h

```diff
@@ -114,6 +119,54 @@ extern "C" {
 #define MBEDTLS_ERR_ERROR_GENERIC_ERROR       -0x0001  /**< Generic error */
 #define MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED -0x006E  /**< This is a bug in the library */
 
+/**
+ * \brief Combines a high-level and low-level error code together.
+ *
+ *        Wrapper function for mbedtls_err_add_ext(). See that function for
+ *        more details.
+ */
+#define MBEDTLS_ERROR_ADD( high, low ) \
+        mbedtls_error_add_ext( high, low, __FILE__, __LINE__ )
+
+#if defined(MBEDTLS_TEST_HOOKS)
+/**
+ * \brief Testing hook called before adding/combining two error codes together.
+ *        Only used when invasive testing is enabled via MBEDTLS_TEST_HOOKS.
+ */
+extern void (*mbedtls_test_hook_error_add)( int, int, const char *, int );
+#endif
+
+/**
+ * \brief Combines a high-level and low-level error code together.
+ *
+ *        This function can be called directly however it is usually
+ *        called via the #MBEDTLS_ERROR_ADD macro.
+ *
+ *        While a value of zero is not a negative error code, it is still an
+ *        error code (that denotes success) and can be combined with both a
+ *        negative error code or another value of zero.
+ *
+ * \note  When invasive testing is enabled via #MBEDTLS_TEST_HOOKS, also try to
+ *        call \link mbedtls_test_hook_error_add \endlink.
+ *
+ * \param high      high-level error code. See error.h for more details.
+ * \param low       low-level error code. See error.h for more details.
+ * \param file      file where this error code addition occured.
+ * \param line      line where this error code addition occured.
+ */
+static inline int mbedtls_error_add_ext( int high, int low,
```

Comment:
Out of curiosity why the "_ext" suffix?

Reply:
See https://github.com/ARMmbed/mbedtls/pull/4006#discussion_r605738001.

Basically I could not think of a better name and it had to be different to `mbedtls_error_add`. If you have a better name please let me know and I'd be happy to change it.

Reply:
Actually now that the macro is capitalised (`MBEDTLS_ERROR_ADD`) I suppose it could be `mbedtls_error_add`. @ronald-cron-arm Would you like me to make this change?

Reply:
Yes please.